nms/app/packages/magmalte/app/views/subscriber/SubscriberDetail.js

```diff
@@ -101,11 +101,23 @@ export default function SubscriberDetail() {
   const classes = useStyles();
   const {relativePath, relativeUrl, match} = useRouter();
   const subscriberId: string = nullthrows(match.params.subscriberId);
+  const ctx = useContext(SubscriberContext);
+  // TODO: render a "Not found" component if the IMSI is not found
+  const subscriberInfo = ctx.state?.[subscriberId] || {};
+
+  let headingTitle = subscriberId;
+  if (
+    subscriberInfo.name !== undefined &&
+    subscriberInfo.name !== null &&
+    subscriberInfo.name.length > 0
+  ) {
+    headingTitle = subscriberInfo.name;
+  }
 
   return (
     <>
       <div className={classes.topBar}>
-        <Text variant="body2">Subscriber/{subscriberId}</Text>
+        <Text variant="body2">Subscriber/{headingTitle}</Text>
```

Comment:
minor nit: You can use null coalescing operator here and write this as 
<Text variant="body2">Subscriber/{subscriberInfo.name ?? subscriberId}</Text>